service/controller/resource/tccpazs/create_test.go

```diff
@@ -639,7 +639,7 @@ func Test_ensureAZsAreAssignedWithSubnet(t *testing.T) {
 	for i, tc := range testCases {
 		t.Run(strconv.Itoa(i), func(t *testing.T) {
 			ctx := controllercontext.NewContext(context.Background(), controllercontext.Context{})
-			azs, err := r.ensureAZsAreAssignedWithSubnet(ctx, tc.tccpSubnet, tc.inputAZs)
+			azs, err := r.ensureAZsAreAssignedWithSubnet(ctx, tc.awsCNISubnet, tc.tccpSubnet, tc.inputAZs)
```

Comment:
looks like line 249 needs and subsequent references to tc.tccpSubnet need to change to tc.awsCNISubnet. hopefully will fix the build

Reply:
The build is broken because we have to fix the tests. We just got the basic implementation sorted how we thought it should work. It wasn't necessary for Pau to watch me fiddle with the unit tests. So this is why I put WIP in the PR name. I need to fix this still.

Reply:
Should be fixed now.